hibernate-core/src/main/java/org/hibernate/resource/jdbc/internal/ResourceRegistryStandardImpl.java

```diff
@@ -69,15 +66,12 @@ public ResourceRegistryStandardImpl(JdbcEventHandler jdbcEventHandler) {
 	@Override
 	public boolean hasRegisteredResources() {
 		return hasRegistered( xref )
-			|| hasRegistered( unassociatedResultSets )
-			|| hasRegistered( blobs )
-			|| hasRegistered( clobs )
-			|| hasRegistered( nclobs );
+			|| ( ext != null && ext.hasRegisteredResources() );
```

Comment:
```suggestion
			|| ext != null && ext.hasRegisteredResources();
```

Reply:
One of the best bits of advice I ever read in a book was: 

> You need to learn the precedence rules of the operators in the language you use every day.

I was like: "oh, yeah, hrrrm, good point".